src/Doctrine/Repository/TaxonRepository.php

```diff
@@ -10,24 +10,90 @@
 
 namespace BitBag\SyliusVueStorefront2Plugin\Doctrine\Repository;
 
+use Doctrine\ORM\EntityRepository;
 use Doctrine\ORM\QueryBuilder;
-use Sylius\Bundle\TaxonomyBundle\Doctrine\ORM\TaxonRepository as BaseTaxonRepository;
+use Gedmo\Tree\Entity\Repository\NestedTreeRepository;
+use Sylius\Bundle\ResourceBundle\Doctrine\ORM\ResourceRepositoryTrait;
 use Sylius\Component\Taxonomy\Model\TaxonInterface;
+use Sylius\Component\Taxonomy\Repository\TaxonRepositoryInterface as BaseTaxonRepositoryInterface;
 
-final class TaxonRepository extends BaseTaxonRepository implements TaxonRepositoryInterface
+final class TaxonRepository extends NestedTreeRepository implements TaxonRepositoryInterface
```

Comment:
Logic in NestedTreeRepository is too complex for merging it without tests. Could you write PHPUnit integration tests for the repository?

Reply:
I wrote a test for a new method. The rest of them is either a call or a copy of the original ones.